src/libsync/creds/httpcredentials.cpp

```diff
@@ -144,6 +151,11 @@ void HttpCredentials::fetchFromKeychain()
     // User must be fetched from config file
     fetchUser();
 
+    if(!_refreshToken.isEmpty()) {
+        refreshAccessToken();
+        return;
```

Comment:
_refreshToken was itself in the keychain.  If we reach this, it means the cached information from the keychain is still valid, but we need to refresh the token in that case.

I updated the comment.

(Yes, that means that fetchFromKeychain is a bad name)